django/db/models/sql/compiler.py

```diff
@@ -938,26 +938,28 @@ def get_converters(self, expressions):
                     converters[i] = (convs, expression)
         return converters
 
-    def apply_converters(self, row, converters):
-        row = list(row)
-        for pos, (convs, expression) in converters.items():
-            value = row[pos]
-            for converter in convs:
-                value = converter(value, expression, self.connection)
-            row[pos] = value
-        return tuple(row)
+    def apply_converters(self, rows, converters):
```

Comment:
@timgraham adding a method call will definitely affect performance on CPython. I think assigning `self` attribute to a local variable is a common pattern to avoid `self.__dict__` lookups in a loop.